### ISSUE TITLE
Table Name Size Violation for cntl_de_thresh.tbl

### ISSUE THREAD

**Issue by JimKaidyNASA** [nasa/SIL#14]

**Comment by JimKaidyNASA** [comment]
The table size max has been increased from 28 to 32 to now 64 as seen in pragmas in the eci_tbl_if.h

![image](https://user-images.githubusercontent.com/52501916/65174613-3cc69780-da1f-11e9-8edc-c3067064e9a4.png)

I don't know if there is another governing parameter I am missing for this table.

![image](https://user-images.githubusercontent.com/52501916/65174563-21f42300-da1f-11e9-86ed-f51c8f01734d.png)

**Comment by SpaceSteve121** [comment]
I believe that the list of relevant table macros is included in [this post](https://github.com/nasa/SIL/issues/11#issuecomment-531373005). 

Can you clarify the intent of this issue? Are you looking for more information on the macros in that list? Do you think there's a problem with the SIL-generated table definition? Are you looking for help parsing an error message?

**Comment by JimKaidyNASA** [comment]
The intent is to seek assistance parsing the error message provided above.
If anything this looks like the earlier issue that we fixed by setting the table max name size.

**Comment by SpaceSteve121** [comment]
As far as I can tell, the images above don't contain an error message... I only see a warning and several notes. If I'm missing it, can you please point it out?

**Comment by JimKaidyNASA** [comment]
Yes they are warnings and I have only circumstantial evidence they are related.  The complaint about the initializer-string  for the array chars is too long seemed to be a hint.  But it's complaining about the ECI_TBL_FILEDEF but ultimately says the recipe for target 'cntl_de_thresh.tbl" failed.  I'm looking for the cause.

**Comment by SpaceSteve121** [comment]
My understanding is that an error is always prefixed with  `error: ` in a gcc log. All I see are lines beginning with `warning: ` and `note: `. I don't see an error anywhere in what you've posted, so I'm struggling because as far as I can tell, you've omitted the actual error that you'd like help interpreting. 

Maybe the error occurred somewhere between the first and second image?

If I'm misunderstanding you, please let me know, but otherwise could you please:
1. Post the entirety of the log from the compilation of the `cntl_de_thresh`?
1. Call out the specific section you'd like help interpreting?

In terms of the warning, it appears to be saying that the space you've allocated for the description is too small to hold the string you've provided. Did you account for the null-terminator and any additions that might be made by CFS?

**Comment by JimKaidyNASA** [comment]
Here is the APP TABLE build listing.  I will look for the gcc log and see if I can put that here.
![image](https://user-images.githubusercontent.com/52501916/65178133-1b1cde80-da26-11e9-9fbd-5ed4f9963e70.png)

**Comment by SpaceSteve121** [comment]
Apologies for the confusion, that is what I meant by the gcc log. I don't see an error in there... perhaps that warning is what's causing the build to fail? If so, perhaps ensure that the length you've allocated is sufficient to hold the description?

**Comment by JimKaidyNASA** [comment]
As before with the srm gains table we ultimately found the parameter and were able to set it to a large enough value for the table name we were using.  That said, this may or may not be related, but I am going to want a similar outcome.  A parameter that somehow governs this size issue that can be set appropriately.

**Comment by SpaceSteve121** [comment]
@JimKaidyNASA That's really more of a CFS configuration question/issue so I'm not sure I can be of much service... perhaps @skliper knows what you're looking for?

**Comment by JimKaidyNASA** [comment]
Looking for some help on why this failure is occurring or at least how to troubleshoot it.
![image](https://user-images.githubusercontent.com/52501916/65266199-9216ad80-dae0-11e9-91ac-74313115468b.png)

**Comment by skliper** [comment]
Likely due to reconfiguring table name sizes, since it works without this change.  Project issues with reconfiguring cFS defaults is typically outside my scope (we've only got resources to best effort support the nominal configuration), is there a cFS expert on your project that can help?  Maybe run a debugger on the elf2cfetbl cntl_de_thresh.o call?

**Comment by JimKaidyNASA** [comment]
We do not have a cFS expert.  Is there a debugger for the build?

**Comment by skliper** [comment]
GDB has worked well for folks in the past.

**Comment by JimKaidyNASA** [comment]
I've used debuggers for after something compiles correctly.

Can you give me a syntactic use of it in place of or part of make?

**Comment by skliper** [comment]
The last failure you shared is on the step where it's trying to run elf2cfetbl on your .o file.  Just replicate that call stand-alone, wrapped with gdb.

**Comment by JimKaidyNASA** [comment]
Sorry to be a pain.  What syntax am I using to replicate that call wrapped with gdb?

**Comment by SpaceSteve121** [comment]
@JimKaidyNASA He's saying that the line 

```
../elf2cfetbl/elf2cfetbl cntl__de_thresh.o
```

which is the first line in the image you posted, is an invocation of the elf2cfetbl tool. Since the problem appears to be coming from there, using GDB to investigate what's happening during the execution of elf2cfetbl might yield additional information.

**Comment by JimKaidyNASA** [comment]
So this?
$gdb ../elf2cfetbl/elf2cftbl cntl__de_thresh.o

**Comment by JimKaidyNASA** [comment]
Not sure how this is working.  The argument being an object file is not what gdb expects (looking for core dump file).
![image](https://user-images.githubusercontent.com/52501916/65276262-5e924e00-daf5-11e9-904f-7b8c7d6f743f.png)

**Comment by SpaceSteve121** [comment]
GDB has extensive documentation which should guide you on how to invoke it properly.

Unfortunately, I don't think the SIL is able to provide any additional support in this regard and so I plan to close this issue. The SIL already attempts to warn you if you're exceeding the CFS settings for table name length, which we've demonstrated work and reflect the changes you made to your CFS configuration. The process for updating those limits, if its going to be supported, would have to be provided by the CFS team. The SIL team, as a single tool within the CFS ecosystem, can't be responsible for developing and documenting those kinds of processes.

Within the scope of the SIL, my suggestion would be to either:
1. Follow the advice of @skliper and try to use CFS-defaults for the table name and description lengths (Note that doing this only requires that you shorten the names of your model and parameter objects in Matlab/Simulink and then regenerate code). 
2. Ensure that your team has the expertise needed to customize the CFS in the ways your project requires.

To try to avoid future users repeating this exercise, I'm going to add a warning to the SIL documentation about the default table name and description lengths, the difficulties involved in changing them, and note that changing them is outside the scope of the SIL's capabilities and documentation. 

If you do find out that this problem is caused by the way the SIL is generating code, or you find a way that the SIL (within its bounds are a CFS app code generator) could better support this use case, please feel free to open another issue and we'll be happy to take a look.

Sorry!

**Comment by JimKaidyNASA** [comment]
Understood Steve.  Thanks for your help.  I will let you know if I discover anything that the SIL might be able to do here.

**Comment by skliper** [comment]
If you need real cFS support there's likely mechanisms to arrange an agreement with GSFC, although something like that is above my level.

**Comment by JimKaidyNASA** [comment]
I've never run a debugger where the argument is an object file.  But even if you helped with this, it would likely lead to other questions.  Thanks @skliper for your help.
